fix: style added for code elements

MAbdurrehman12 opened the pull request:
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [ ] Mobile width screenshots
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
https://github.com/mitodl/ocw-to-hugo/issues/464

#### What's this PR do?
- Overrides default css styling of `code` elements

#### How should this be manually tested?
- Checkout to the branch of [this PR](https://github.com/mitodl/ocw-to-hugo/pull/469)
- Run any course through it (course which has one liner codes and code blocks), [example course](https://ocwnext.odl.mit.edu/courses/6-006-introduction-to-algorithms-fall-2011/)
- Build this course in `ocw-hugo-themes` 
- Verify that all the single line codes and code blocks have appropriate styling
- Verify that this change has no `side-effect` anywhere else 
- Repeat this step for multiple courses and multiple browsers

#### Screenshots (if appropriate)

![image](https://user-images.githubusercontent.com/93309234/155318478-6a0bcff0-4055-4db5-b8c2-60ea6ed4eaea.png)

![image](https://user-images.githubusercontent.com/93309234/155318541-a51d9a8d-9c01-4fee-96aa-8966ce73de8a.png)

![image](https://user-images.githubusercontent.com/93309234/155318394-294715f9-64d3-4b9a-8586-51364c7b6cc8.png)

MAbdurrehman12 commented:
I think, all the code sections (single line code and code blocks) should have a `background color` and a `button`, at right most side or top right corner, to copy that code snippet to clipboard. Currently students will have to manually copy the code.
I guess this is doable via `javascript/jquery` in this repo (ocw-hugo-themes) but not sure how much value it will add or will it be worth the effort.
@pdpinch Do you have some thoughts on this?

pdpinch commented:
I'd rather keep this plan and simple for now: monospaced font family is enough.

MAbdurrehman12 commented:
> I'd rather keep this plan and simple for now: monospaced font family is enough. 


Okay, then I guess we can get this PR reviewed.
The current font-family applied is from monospaced family

MAbdurrehman12 commented:
> I'd rather keep this plan and simple for now: monospaced font family is enough. 


Okay, then I guess we can get this PR reviewed.
The current font-family applied is from monospaced family

pdpinch commented:
unimportant nit: I would call this a "feat" in the commit message, not "fix"

mbertrand reviewed (commented):
On page http://localhost:3000/pages/readings/python-cost-model/, the code block text is empty:
<img width="637" alt="Screen Shot 2022-03-02 at 3 56 59 PM" src="https://user-images.githubusercontent.com/187676/156452597-9d8ecd9a-efea-4bc2-9f06-7b93921e7809.png">

I think this is due more to the [related ocw-to-hugo PR though](https://github.com/mitodl/ocw-to-hugo/pull/469), when I switch to the main branch of ocw-to-hugo it's no longer blank:
<img width="626" alt="Screen Shot 2022-03-02 at 4 03 05 PM" src="https://user-images.githubusercontent.com/187676/156452621-32f1aeb8-385f-4543-9e3f-2936550c978c.png">

MAbdurrehman12 commented:
@mbertrand Thanks for looking into this.
Yes, for now, to test this, we'll have to checkout to `[abdurrehman/code-formatting](https://github.com/mitodl/ocw-to-hugo/tree/abdurrehman/code-formatting)` branch of `ocw-to-hugo` as turndown rules for code snippets are added in this [PR](https://github.com/mitodl/ocw-to-hugo/pull/469).

I think, we'll merge this, once the `ocw-to-hugo` PR is merged.

MAbdurrehman12 commented:
@mbertrand Sorry, I misunderstood your comment initially.
Thank you for pointing this out.
After looking into it, it looks like another use-case, i.e: strings that starts with `tab` are automatically converted into code block while Markdown to HTML so I guess we need to ignore such strings.
I have fixed it in the [ocw-to-hugo PR](https://github.com/mitodl/ocw-to-hugo/pull/469) and fixed another small issue of leading backslash.
Can you please have another look?

mbertrand reviewed (approved):
👍
